.cursor/rules/responsive-design-patterns.mdc

```diff
@@ -0,0 +1,191 @@
+---
+alwaysApply: true
+---
+
+# Responsive Design: Container Queries Over Media Queries
+
+## Principle
+
+Use container-based styling and `useContainerBreakpoints` for all responsive behavior. Since the SDK is embedded in various contexts and container sizes, component behavior must adapt to the component's actual container, not the viewport. This is essential for predictable behavior across different host applications.
```

Comment:
This one is incorrect. We should use `Grid` and `Flex` components with responsive props. Those components have native css container queries baked in which will be more performant than useContainerBreakpoints which uses resize observer. `useContainerBreakpoints` should only be used when we need to make significant changes to the rendering of a layout that go beyond just adjusting the grid or flex containers. Ex. document signer has a completely different document preview treatment on mobile